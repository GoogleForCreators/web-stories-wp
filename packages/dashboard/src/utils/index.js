/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export { default as titleFormatted } from './titleFormatted';
export { default as KeyboardOnlyOutline } from './keyboardOnlyOutline';
export { default as useDashboardResultsLabel } from './useDashboardResultsLabel';
export {
  default as usePagePreviewSize,
  getPosterHeight,
} from './usePagePreviewSize';
export { default as useStoryView } from './useStoryView';
export { default as useTemplateView } from './useTemplateView';
export { noop } from './noop';
