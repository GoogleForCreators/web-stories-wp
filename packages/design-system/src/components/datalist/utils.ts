/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import type { Group, AbstractOption } from './types';

export const createOptionFilter =
  <O extends AbstractOption>(options: O[]) =>
  (keyword: string) =>
    options.filter(({ name }) =>
      name.toLowerCase().includes(keyword.toLowerCase())
    );

export const isKeywordFilterable = (keyword: string) =>
  keyword.trim().length >= 2;

export const getOptions = <O extends AbstractOption>(groups: Group<O>[]) =>
  groups.flatMap(({ options }) => options);

export const addUniqueEntries = <T>(array: T[], ...keys: T[]) => [
  ...new Set(array.concat(keys)),
];

export const getInset = <O extends AbstractOption>(
  groups: Group<O>[],
  i: number,
  j: number
) =>
  groups
    .slice(0, i)
    .map((group) => group.options.length)
    .reduce((a, b) => a + b, 0) + j;
