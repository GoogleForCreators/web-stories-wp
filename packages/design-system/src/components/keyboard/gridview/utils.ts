/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// @todo: provide a cleaner `focusFirst()` API and takes into account
// `tabIndex`, `disabled`, links, buttons, inputs, etc.
export function focusOnPage(page: Element | undefined) {
  // grab the first focusable element available.
  // currently can be a button or an anchor.
  const button = page?.querySelector('button, a') as
    | HTMLButtonElement
    | HTMLAnchorElement;
  if (button) {
    button.focus();
  }
}

export function getArrowDir(
  key: string,
  pos: string,
  neg: string,
  isRTL: boolean
) {
  const rtlDir = isRTL ? -1 : 1;
  if (key === pos) {
    return rtlDir;
  }
  if (key === neg) {
    return -1 * rtlDir;
  }
  return 0;
}

export function getGridColumnAndRowCount(grid: HTMLElement, pageCount: number) {
  let columns = 0;
  let prevX;
  for (const el of grid.children) {
    const { x } = el.getBoundingClientRect();
    if (typeof prevX !== 'undefined' && x < prevX) {
      break;
    }
    prevX = x;
    columns++;
  }

  const rows = Math.ceil(pageCount / columns);

  return { rows, columns };
}

// will return a 1 based index
export function getRow(index: number, numColumns: number) {
  return Math.ceil((index + 1) / numColumns);
}

// will return a 1 based index
export function getColumn(index: number, numColumns: number) {
  return (index % numColumns) + 1;
}

// will return a 0 based index
export function getIndex({
  row,
  column,
  numRows,
  numColumns,
  numItems,
}: {
  row: number;
  column: number;
  numRows: number;
  numColumns: number;
  numItems: number;
}) {
  const isOutOfBounds =
    row > numRows || row <= 0 || column > numColumns || column <= 0;

  if (isOutOfBounds) {
    return -1;
  }

  const index = numColumns * (row - 1) + (column - 1);

  // If the index is greater than or equal to numItems default to the last index.
  // This handles the case when we press ArrowDown and there is another row, but
  // the column below is empty. It will default to the last item in the list.
  return index >= numItems ? numItems - 1 : index;
}
