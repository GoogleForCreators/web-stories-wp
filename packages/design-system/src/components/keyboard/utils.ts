/*
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { __ } from '@googleforcreators/i18n';

/**
 * Internal dependencies
 */
import Mousetrap from 'mousetrap';
import type { MousetrapInstance } from 'mousetrap';
import type {
  RefOrNode,
  KeyEffectCallback,
  KeyNameOrSpec,
  Keys,
} from '../../types/keyboard';

const PROP = '__WEB_STORIES_MT__';
const NON_EDITABLE_INPUT_TYPES = [
  'submit',
  'button',
  'checkbox',
  'radio',
  'image',
  'file',
  'range',
  'reset',
  'hidden',
];
const CLICKABLE_INPUT_TYPES = [
  'submit',
  'button',
  'checkbox',
  'radio',
  'image',
  'file',
  'reset',
];

export interface HTMLElementWithMouseTrap extends HTMLElement {
  [PROP]: undefined | MousetrapInstance;
}
/**
 * @param node The DOM node.
 * @return The Mousetrap object that will be used to intercept
 * the keyboard events on the specified node.
 */
export function getOrCreateMousetrap(node: HTMLElementWithMouseTrap) {
  return node[PROP] || (node[PROP] = new Mousetrap(node));
}

export function getNodeFromRefOrNode(refOrNode: RefOrNode) {
  return refOrNode && 'current' in refOrNode ? refOrNode.current : refOrNode;
}

export function resolveKeySpec(keyDict: Keys, keyNameOrSpec: KeyNameOrSpec) {
  const keySpec =
    typeof keyNameOrSpec === 'string' || Array.isArray(keyNameOrSpec)
      ? { key: keyNameOrSpec }
      : keyNameOrSpec;
  const {
    key: keyOrArray,
    shift = false,
    repeat = true,
    clickable = true,
    editable = false,
    dialog = false,
    allowDefault = false,
  } = keySpec;
  const mappedKeys = new Array<string>()
    .concat(keyOrArray)
    .map((key) => keyDict[key as keyof Keys] || key)
    .flat();
  const allKeys = addMods(mappedKeys, shift);
  return {
    key: allKeys,
    shift,
    clickable,
    repeat,
    editable,
    dialog,
    allowDefault,
  };
}

export function addMods(keys: string[], shift: boolean) {
  if (!shift) {
    return keys;
  }
  return keys.concat(keys.map((key) => `shift+${key}`));
}

interface KeyHandlerProps {
  repeat?: boolean;
  editable?: boolean;
  clickable?: boolean;
  dialog?: boolean;
  allowDefault?: boolean;
}

export function createKeyHandler(
  keyTarget: Element,
  {
    repeat: repeatAllowed,
    editable: editableAllowed,
    clickable: clickableAllowed,
    dialog: dialogAllowed,
    allowDefault = false,
  }: KeyHandlerProps,
  callback: KeyEffectCallback
) {
  return (evt: KeyboardEvent) => {
    const { repeat, target } = evt;
    if (!repeatAllowed && repeat) {
      return undefined;
    }
    if (!editableAllowed && isEditableTarget(target as HTMLInputElement)) {
      return undefined;
    }
    if (!clickableAllowed && isClickableTarget(target as HTMLInputElement)) {
      return undefined;
    }
    if (
      !dialogAllowed &&
      crossesDialogBoundary(target as HTMLElement, keyTarget)
    ) {
      return undefined;
    }
    callback(evt);
    // The default `false` value instructs Mousetrap to cancel event propagation
    // and default behavior.
    return allowDefault;
  };
}

type ClickableHTMLElement =
  | HTMLInputElement
  | HTMLAnchorElement
  | HTMLButtonElement
  | HTMLTextAreaElement;
export function isClickableTarget({ tagName, type }: ClickableHTMLElement) {
  if (['BUTTON', 'A'].includes(tagName)) {
    return true;
  }
  if (tagName === 'INPUT') {
    return CLICKABLE_INPUT_TYPES.includes(type);
  }
  return false;
}

export function isEditableTarget({
  tagName,
  isContentEditable,
  type,
  ...rest
}: ClickableHTMLElement) {
  if ('readOnly' in rest && rest.readOnly === true) {
    return false;
  }
  if (isContentEditable || tagName === 'TEXTAREA') {
    return true;
  }
  if (tagName === 'INPUT') {
    return !NON_EDITABLE_INPUT_TYPES.includes(type);
  }
  return false;
}

export function crossesDialogBoundary(target: Element, keyTarget: Element) {
  if (target.nodeType !== 1) {
    // Not an element. Most likely a document node. The dialog search
    // does not apply.
    return false;
  }
  // Check if somewhere between `keyTarget` and `target` there's a
  // dialog boundary.
  const dialog = target.closest('dialog,[role="dialog"]');
  return dialog && keyTarget !== dialog && keyTarget.contains(dialog);
}

/**
 * Determines if the current platform is a Mac or not.
 */
export function isPlatformMacOS() {
  const { platform } = window.navigator;
  return platform.includes('Mac') || ['iPad', 'iPhone'].includes(platform);
}

/**
 * Get the key specific to operating system.
 */
export function getKeyForOS(key: string) {
  const isMacOS = isPlatformMacOS();

  const replacementKeyMap: Record<string, string> = {
    alt: isMacOS ? '⌥' : 'Alt',
    ctrl: isMacOS ? '^' : 'Ctrl',
    mod: isMacOS ? '⌘' : 'Ctrl',
    cmd: '⌘',
    shift: isMacOS ? '⇧' : 'Shift',
  };

  return replacementKeyMap[key] || key;
}

/**
 * Creates an aria label for a shortcut.
 *
 * Inspired from the worpress Gutenberg plugin:
 * https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/keycodes/src/index.js#L240-L277
 *
 * @example
 * createShortcutAriaLabel('mod alt del'); -> "Command Alt Del"
 */
export function createShortcutAriaLabel(shortcut: string) {
  const isMacOS = isPlatformMacOS();

  /* translators: Command key on the keyboard */
  const command = __('Command', 'web-stories');
  /* translators: Control key on the keyboard */
  const control = __('Control', 'web-stories');
  /* translators: Option key on the keyboard */
  const option = __('Option', 'web-stories');
  /* translators: Alt key on the keyboard */
  const alt = __('Alt', 'web-stories');

  const replacementKeyMap = {
    alt: isMacOS ? option : alt,
    mod: isMacOS ? command : control,
    /* translators: Control key on the keyboard */
    ctrl: __('Control', 'web-stories'),
    /* translators: shift key on the keyboard */
    shift: __('Shift', 'web-stories'),
    /* translators: delete key on the keyboard */
    delete: __('Delete', 'web-stories'),
    cmd: command,
    /* translators: comma character ',' */
    ',': __('Comma', 'web-stories'),
    /* translators: period character '.' */
    '.': __('Period', 'web-stories'),
    /* translators: backtick character '`' */
    '`': __('Backtick', 'web-stories'),
  };

  const delimiter = isMacOS ? ' ' : '+';

  return shortcut
    .toLowerCase()
    .replace('alt', replacementKeyMap.alt)
    .replace('ctrl', replacementKeyMap.ctrl)
    .replace('mod', replacementKeyMap.mod)
    .replace('cmd', replacementKeyMap.cmd)
    .replace('shift', replacementKeyMap.shift)
    .replace('delete', replacementKeyMap.delete)
    .replace(',', replacementKeyMap[','])
    .replace('.', replacementKeyMap['.'])
    .replace('`', replacementKeyMap['`'])
    .split(/[\s+]/)
    .map((s) => s.charAt(0).toUpperCase() + s.slice(1))
    .join(delimiter);
}

/**
 * Prettifies keyboard shortcuts in a platform-agnostic way.
 */
export function prettifyShortcut(shortcut: string) {
  const isMacOS = isPlatformMacOS();

  const delimiter = isMacOS ? '' : '+';

  return shortcut
    .toLowerCase()
    .replace('alt', getKeyForOS('alt'))
    .replace('ctrl', getKeyForOS('ctrl'))
    .replace('mod', getKeyForOS('mod'))
    .replace('cmd', getKeyForOS('cmd'))
    .replace('shift', getKeyForOS('shift'))
    .replace('left', '←')
    .replace('up', '↑')
    .replace('right', '→')
    .replace('down', '↓')
    .replace('delete', '⌫')
    .replace('enter', '⏎')
    .split('+')
    .map((s) => s.charAt(0).toUpperCase() + s.slice(1))
    .join(delimiter);
}
