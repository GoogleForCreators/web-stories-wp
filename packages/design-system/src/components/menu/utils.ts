/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import type { DropdownItem, NestedDropdownItem } from './types';

function isValid(opt: DropdownItem) {
  return typeof opt === 'object' && typeof opt.value !== 'undefined';
}

export function getGroups(
  source: NestedDropdownItem[] | DropdownItem[]
): NestedDropdownItem[] {
  if (source.length === 0) {
    return [];
  }
  const isNested = source.some(
    (opt) => typeof opt === 'object' && 'options' in opt
  );
  if (isNested) {
    const nestedOptions = source as NestedDropdownItem[];
    return nestedOptions
      .map((group) => ({
        ...group,
        options: (group.options || []).filter(isValid),
      }))
      .filter(({ options }) => options.length > 0);
  }

  // Double check that all data in groups should be treated as options to sanitize data
  const options = (source as DropdownItem[]).filter(isValid);
  return options.length ? [{ options }] : [];
}

export function getInset(
  groups: NestedDropdownItem[],
  i: number,
  j: number
): number {
  return (
    groups
      .slice(0, i)
      .map(({ options }) => options.length)
      .reduce((a: number, b: number) => a + b, 0) + j
  );
}
