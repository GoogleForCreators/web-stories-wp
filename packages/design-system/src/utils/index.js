/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export { default as isNullOrUndefinedOrEmptyString } from './isNullOrUndefinedOrEmptyString';
export { noop } from './noop';
export { default as addQueryArgs } from './addQueryArgs';
export { default as labelAccessibilityValidator } from './labelAccessibilityValidator';
export { default as useLiveRegion } from './useLiveRegion';
export { default as uniqueEntriesByKey } from './uniqueEntriesByKey';
export { default as domReady } from './domReady';
export { default as deepMerge } from './deepMerge';
export { default as useMouseDownOutsideRef } from './useMouseDownOutsideRef';
export * from './constants';
export * from './directions';
export * from './url';
