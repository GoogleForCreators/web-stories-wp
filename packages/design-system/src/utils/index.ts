/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export { default as isNullOrUndefinedOrEmptyString } from './isNullOrUndefinedOrEmptyString';
export { noop } from './noop';
export { default as labelAccessibilityValidator } from './labelAccessibilityValidator';
export { default as useLiveRegion } from './useLiveRegion';
export { default as deepMerge } from './deepMerge';
export { default as useMouseDownOutsideRef } from './useMouseDownOutsideRef';
export { default as getKeyboardMovement } from './getKeyboardMovement';
export { default as uniqueEntriesByKey } from './uniqueEntriesByKey';
export { default as PanelTypes } from './panelTypes';
export { PanelSections } from './panelSections';
export { STYLE_PANE_IDS } from './panelSections';
export * from './constants';
export * from './directions';
