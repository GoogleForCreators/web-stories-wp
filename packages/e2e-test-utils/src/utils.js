/*
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * External dependencies
 */
import { getDocument, queries } from 'pptr-testing-library';

export const getByTestId = async (page, testId) => {
  const $document = await getDocument(page);
  return queries.getByTestId($document, testId);
};

export const getByAltText = async (page, testId) => {
  const $document = await getDocument(page);
  return queries.getByAltText($document, testId);
};

export const getByText = async (page, testId) => {
  const $document = await getDocument(page);
  return queries.getByText($document, testId);
};

export const getByLabelText = async (page, testId) => {
  const $document = await getDocument(page);
  return queries.getByLabelText($document, testId);
};
