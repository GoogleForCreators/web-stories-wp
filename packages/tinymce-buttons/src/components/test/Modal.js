/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * External dependencies
 */
import React from 'react';
import renderer from 'react-test-renderer';
/**
 * Internal dependencies
 */
import { isView } from '../../utils';
import WebStoriesModal from '../Modal';

jest.mock('@wordpress/data', () => ({
  select: jest.fn(() => ({
    getCurrentView: () => 'grid',
  })),
}));

jest.mock('@wordpress/i18n', () => ({
  // eslint-disable-next-line no-unused-vars
  __: (val, domain) => val,
}));

jest.mock('@wordpress/components', () => ({
  Modal: 'Modal',
  RangeControl: 'Range',
  SelectControl: 'Select',
  Button: 'Button',
  TextControl: 'TextControl',
}));

jest.mock('../../utils/globals', () => ({
  webStoriesData: {
    views: [],
    orderlist: [],
  },
}));

jest.mock('../../components/controls/Toggle', () => 'TinyMCEToggle');

jest.mock('../../utils', () => ({
  isView: jest.fn((view) => view === 'grid'),
  isCircleView: jest.fn(() => false),
  updateViewSettings: jest.fn((settings) => settings),
}));

describe('TinyMCE Controls Modal', () => {
  it('modal is opened and view is non-circle', () => {
    const props = {
      modalOpen: true,
      settings: {
        archive_link: {
          show: true,
        },
      },
      prepareShortCode: jest.fn(),
    };

    const tree = renderer.create(<WebStoriesModal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('modal is opened and view is circle', () => {
    isView.mockImplementationOnce((view) => view === 'circles');

    const props = {
      modalOpen: true,
      settings: {
        archive_link: {
          show: true,
        },
      },
      prepareShortCode: jest.fn(),
    };

    const tree = renderer.create(<WebStoriesModal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
