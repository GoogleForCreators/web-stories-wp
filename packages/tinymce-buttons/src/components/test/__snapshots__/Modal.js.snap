// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TinyMCE Controls Modal modal is opened and view is circle 1`] = `
<Modal
  className="component_web_stories_mce_model"
  closeButtonLabel="Close"
  onRequestClose={[Function]}
  shouldCloseOnClickOutside={false}
  title="Web Stories"
>
  <Select
    label="Select View Type"
    onChange={[Function]}
    options={Array []}
  />
  <TinyMCEToggle
    field="title"
  />
  <TinyMCEToggle
    field="excerpt"
  />
  <TinyMCEToggle
    field="author"
  />
  <TinyMCEToggle
    field="date"
  />
  <TinyMCEToggle
    field="image_align"
  />
  <TinyMCEToggle
    field="sharp_corners"
  />
  <TinyMCEToggle
    field="archive_link"
    fieldObj={
      Object {
        "show": true,
      }
    }
  />
  <TextControl
    label="Archive Link Label"
    onChange={[Function]}
  />
  <Range
    label="Number of Stories"
    max={20}
    min={1}
    onChange={[Function]}
  />
  <Range
    label="Circle Size"
    max={200}
    min={80}
    onChange={[Function]}
    step={5}
  />
  <Range
    label="Number of Columns"
    max={4}
    min={1}
    onChange={[Function]}
  />
  <Select
    label="Select Order"
    onChange={[Function]}
    options={Array []}
  />
  <div
    className="alignright"
    style={
      Object {
        "padding": "20px 0",
      }
    }
  >
    <Button
      isPrimary={true}
      onClick={[Function]}
    >
      Okay
    </Button>
    <Button
      onClick={[Function]}
    >
      Cancel
    </Button>
  </div>
</Modal>
`;

exports[`TinyMCE Controls Modal modal is opened and view is non-circle 1`] = `
<Modal
  className="component_web_stories_mce_model"
  closeButtonLabel="Close"
  onRequestClose={[Function]}
  shouldCloseOnClickOutside={false}
  title="Web Stories"
>
  <Select
    label="Select View Type"
    onChange={[Function]}
    options={Array []}
  />
  <TinyMCEToggle
    field="title"
  />
  <TinyMCEToggle
    field="excerpt"
  />
  <TinyMCEToggle
    field="author"
  />
  <TinyMCEToggle
    field="date"
  />
  <TinyMCEToggle
    field="image_align"
  />
  <TinyMCEToggle
    field="sharp_corners"
  />
  <TinyMCEToggle
    field="archive_link"
    fieldObj={
      Object {
        "show": true,
      }
    }
  />
  <TextControl
    label="Archive Link Label"
    onChange={[Function]}
  />
  <Range
    label="Number of Stories"
    max={20}
    min={1}
    onChange={[Function]}
  />
  <Range
    label="Number of Columns"
    max={4}
    min={1}
    onChange={[Function]}
  />
  <Select
    label="Select Order"
    onChange={[Function]}
    options={Array []}
  />
  <div
    className="alignright"
    style={
      Object {
        "padding": "20px 0",
      }
    }
  >
    <Button
      isPrimary={true}
      onClick={[Function]}
    >
      Okay
    </Button>
    <Button
      onClick={[Function]}
    >
      Cancel
    </Button>
  </div>
</Modal>
`;
